src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable {
+    public static final int MAX_SIZE = 100;
+    private Color color;
+
+    public Figure(Color color) {
+        this.color = color;
+    }
+
+    public Color getColor() {
+        return color;
+    }
+
+    public void setColor(Color color) {
+        this.color = color;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: " + getShape().getDescription().toLowerCase()
+                + ", area: " + String.format("%.2f", getArea()) + " sq.units, "
+                + getSizeDescription() + ", color: " + color.name().toLowerCase());
+    }
+
+    public abstract double getArea();
+
+    public abstract Shape getShape();
+
+    protected abstract String getSizeDescription();
```

Comment:
why access modifer different here?

Reply:
getShape() represents a type of figure and could be considered the Figure class property.
getSizeDescription() was added as an utility method for draw() implementation. It wasn't supposed to be used independently. It might be called like describeSizes() to prevent confusing it with a getter.
Well, it could be both public, if we treat getSizeDescription() as a class property following getter's convention.
Let's go with the public. 
Thanks!